File: backend/app/schemas/image.py
from pydantic import BaseModel
from typing import Any, Dict, Optional
from app.db.models import ProductConditionEnum


class ImageUploadResponse(BaseModel):
    image_key: str
    image_url: str
    suggested_title: str
    suggested_category_id: Optional[int] = None
    suggested_attributes: Optional[Dict[str, Any]] = None
    suggested_description: Optional[str] = None
    suggested_condition: Optional[ProductConditionEnum] = None


class ImageDeleteResponse(BaseModel):
    message: str
    image_key: str

File: frontend/src/types/image.ts
import { productConditionEnum } from "./product";

export interface Image {
  image_key: string;
  image_url: string;
  suggested_title?: string;
  suggested_attributes: Record<string, string>;
  suggested_category_id?: number;
  suggested_description?: string;
  suggested_condition?: typeof productConditionEnum[number];
}